Installation/cmake/modules/CGAL_pointmatcher_support.cmake

```diff
@@ -1,7 +1,19 @@
 if(libpointmatcher_FOUND AND NOT TARGET CGAL::pointmatcher_support)
-  add_library(CGAL::pointmatcher_support INTERFACE IMPORTED)
-  set_target_properties(CGAL::pointmatcher_support PROPERTIES
-    INTERFACE_COMPILE_DEFINITIONS "CGAL_LINKED_WITH_POINTMATCHER"
-    INTERFACE_INCLUDE_DIRECTORIES "${libpointmatcher_INCLUDE_DIR}"
-    INTERFACE_LINK_LIBRARIES "${libpointmatcher_LIBRARIES}")
+  if(WIN32 OR CMAKE_SYSTEM_NAME STREQUAL Windows)
+    find_package(Boost COMPONENTS thread filesystem system program_options date_time chrono)
+  endif()
+  if(NOT (WIN32 OR CMAKE_SYSTEM_NAME STREQUAL Windows)
+      OR ( Boost_chrono_FOUND
+      AND Boost_thread_FOUND
+      AND Boost_filesystem_FOUND
+      AND Boost_system_FOUND
+      AND Boost_program_options_FOUND
+      AND Boost_date_time_FOUND) )
+    add_library(CGAL::pointmatcher_support INTERFACE IMPORTED)
+    target_compile_definitions(CGAL::pointmatcher_support INTERFACE "CGAL_LINKED_WITH_POINTMATCHER")
+    target_include_directories(CGAL::pointmatcher_support INTERFACE "${libpointmatcher_INCLUDE_DIR}")
+    target_link_libraries(CGAL::pointmatcher_support INTERFACE "${libpointmatcher_LIBRARIES}")
+  else()
+    message(STATUS "NOTICE : the libpointmatcher library requires the following boost components: thread filesystem system program_options date_time chrono.")
+  endif()
```

Comment:
At the current time, either we fix the support in our code, either we completely drop the support of the Windows platforms. I already reported the bugs in Pointmatcher's cmake scripts [here](https://github.com/ethz-asl/libpointmatcher/issues/435)

Reply:
Also, we don't need it to work with the next version, as it is documented that we do not support another version. At least until they fix the bugs we reported.

Reply:
> Also, we don't need it to work with the next version, as it is documented that we do not support another version. At least until they fix the bugs we reported.

I know that is documented that way, but that is very aggressive for our users. Imagine somebody said "we only support CGAL version 4.13.2"... That is why, if that library is important for us (@sgiraudot?) then we should collaborate with `libpointmatcher` to improve their CMake support. And I think we should contribute pull-requests, and not only issues, against if that is important for us.

Reply:
It's really "just" a wrapper.

It's important in the sense that it is something that we do provide and thus we should make sure it works, but I'm not sure how much time and effort it makes sense to put into that (especially if they have an almost non-existent Windows support, it might be a long run to make it clean).

Reply:
> It's really "just" a wrapper.
> 
> It's important in the sense that it is something that we do provide and thus we should make sure it works, but I'm not sure how much time and effort it makes sense to put into that (especially if they have an almost non-existent Windows support, it might be a long run to make it clean).

The same way we helped Eigen3 with their Windows support, maybe we could help `libpointmatcher` with their Windows support. I do not know how much effort we can put, though. We should ask Andreas (@afabri).